solc/jsonCompiler.cpp

```diff
@@ -203,51 +203,65 @@ string compile(StringMap const& _sources, bool _optimize, CStyleReadFileCallback
 
 	if (success)
 	{
-		output["contracts"] = Json::Value(Json::objectValue);
-		for (string const& contractName: compiler.contractNames())
+		try
 		{
-			Json::Value contractData(Json::objectValue);
-			contractData["interface"] = compiler.interface(contractName);
-			contractData["bytecode"] = compiler.object(contractName).toHex();
-			contractData["runtimeBytecode"] = compiler.runtimeObject(contractName).toHex();
-			contractData["opcodes"] = solidity::disassemble(compiler.object(contractName).bytecode);
-			contractData["functionHashes"] = functionHashes(compiler.contractDefinition(contractName));
-			contractData["gasEstimates"] = estimateGas(compiler, contractName);
-			auto sourceMap = compiler.sourceMapping(contractName);
-			contractData["srcmap"] = sourceMap ? *sourceMap : "";
-			auto runtimeSourceMap = compiler.runtimeSourceMapping(contractName);
-			contractData["srcmapRuntime"] = runtimeSourceMap ? *runtimeSourceMap : "";
-			ostringstream unused;
-			contractData["assembly"] = compiler.streamAssembly(unused, contractName, _sources, true);
-			output["contracts"][contractName] = contractData;
-		}
+			output["contracts"] = Json::Value(Json::objectValue);
+			for (string const& contractName: compiler.contractNames())
+			{
+				Json::Value contractData(Json::objectValue);
+				contractData["interface"] = compiler.interface(contractName);
+				contractData["bytecode"] = compiler.object(contractName).toHex();
+				contractData["runtimeBytecode"] = compiler.runtimeObject(contractName).toHex();
+				contractData["opcodes"] = solidity::disassemble(compiler.object(contractName).bytecode);
+				contractData["functionHashes"] = functionHashes(compiler.contractDefinition(contractName));
+				contractData["gasEstimates"] = estimateGas(compiler, contractName);
+				auto sourceMap = compiler.sourceMapping(contractName);
+				contractData["srcmap"] = sourceMap ? *sourceMap : "";
+				auto runtimeSourceMap = compiler.runtimeSourceMapping(contractName);
+				contractData["srcmapRuntime"] = runtimeSourceMap ? *runtimeSourceMap : "";
+				ostringstream unused;
+				contractData["assembly"] = compiler.streamAssembly(unused, contractName, _sources, true);
+				output["contracts"][contractName] = contractData;
+			}
+
+			// Do not taint the internal error list
+			ErrorList formalErrors;
+			if (compiler.prepareFormalAnalysis(&formalErrors))
+				output["formal"]["why3"] = compiler.formalTranslation();
```

Comment:
Sure, but with the same argument, we could add a try/catch block around every single line. Compiler bugs are easy to reproduce and a debugger will show you the exact position of the problem. The benefit to the user could be to switch off the problematic part in the output, but so far this has only happened once.